ledger/error/src/main/scala/com/daml/error/ErrorCode.scala

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.error
+
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import io.grpc.StatusRuntimeException
+import org.slf4j.event.Level
+
+import scala.annotation.StaticAnnotation
+import scala.util.control.NoStackTrace
+
+/** Error Code Definition
```

Comment:
Would be maybe good to start this doc with an explanation where `ErrorCode` fits within the structure of error related classes. E.g.:
"""
Error codes gathers errors with the same error code.
Error code belongs to an error group.
Error codes determines high level details what it is and how to deal with it. For more information refer to particular subclass of BaseError.
"""

Reply:
Let's leave this for a refactoring PR as well.